src/org/zaproxy/zap/model/DefaultValueGenerator.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2016 The ZAP Development Team
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.model;
+
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+
+import org.apache.commons.httpclient.URI;
+
+import net.htmlparser.jericho.FormControlType;
+
+/**
+ *   Default implementation of the DefaultValueGenerator
+ *   
+ *   @since 2.6.0
+ *   @return the default String value for each control type
+ */
+
+public class DefaultValueGenerator implements ValueGenerator{
+
+	private static final String ATTR_TYPE = "type";
+	private static final String DEFAULT_NUMBER_VALUE = "1";
+	private static final String DEFAULT_TEXT_VALUE = org.parosproxy.paros.Constant.PROGRAM_NAME_SHORT;
+	private static final String DEFAULT_PASS_VALUE = DEFAULT_TEXT_VALUE;
+	private static final String DEFAULT_FILE_VALUE = "test_file.txt";
+	private static final String DEFAULT_EMPTY_VALUE = "";
+
+	private Date defaultDate;
+	
+	/**
+	 * Gets the default {@code Date}, to be used for default values of date fields.
+	 *
+	 * @return the date, never {@code null}.
+	 * @see #setDefaultDate(Date)
+	 */
+	public Date getDefaultDate() {
+		if (defaultDate == null) {
+			return new Date();
+		}
+		return defaultDate;
+	}
+
+	public void setDefaultDate(Date date) {
+		this.defaultDate = date;
+	}
+
+	/**
+	 * Generates accurate field values for following types:
+	 * <ul>
+	 * <li>Text/Password/Search - DEFAULT_TEXT_VALUE</li>
+	 * <li>number/range - if min is defined, then use min. If max is defined use max,
+	 * otherwise DEFAULT_NUMBER_VALUE</li>
+	 * <li>url - http://www.example.com</li>
+	 * <li>email - foo-bar@example.com</li>
+	 * <li>color - #ffffff</li>
+	 * <li>tel - 9999999999</li>
+	 * <li>date/datetime/time/month/week/datetime-local - current date in the proper format</li>
+	 * <li>file - DEFAULT_FILE_VALUE</li>
+	 * </ul> 
+	 */
+	@Override
+	public String getValue(URI uri, String url, String fieldId, String defaultValue, Map<String, String> envAttributes, Map<String, String> fieldAttributes) {
+
+		//If there is a default value provided, return it
+		if (defaultValue != ""){
+			return defaultValue;

```

Comment:
This conditional should return the same default value, however our extension should be able to modify the value now that it's passed in.